procrastinate/worker.py

```diff
@@ -131,7 +130,7 @@ async def _persist_job_status(
                 jobs.DeleteJobCondition.ALWAYS: True,
                 jobs.DeleteJobCondition.NEVER: False,
                 jobs.DeleteJobCondition.SUCCESSFUL: status == jobs.Status.SUCCEEDED,
-            }[self.delete_jobs]
+            }[self.delete_jobs or jobs.DeleteJobCondition.NEVER]
```

Comment:
looks like we could do it on line 57 and avoid having the attribute be nullable. Just a suggestion. Also, it's unclear why it was changed so maybe I'm missing the point.